components/tidb_query/src/rpn_expr/impl_compare.rs

```diff
@@ -207,10 +207,46 @@ pub fn coalesce<T: Evaluable>(args: &[&Option<T>]) -> Result<Option<T>> {
     Ok(None)
 }
 
+#[inline]
+fn do_get_extremum<F, T>(list: &[&Option<T>], chooser: F) -> Result<Option<T>>
```

Comment:
Looks like we can extract metadata for greatest and least? @TennyZhuang

Reply:
Yes, we can extract greatest and least in constants and save it in metadata
.

Reply:
Hi @Aloxaf , thanks for the contribution! According to @TennyZhuang this PR can be optimized by extracting the extremum constant value beforehand. However currently we are refining related part. You may wait https://github.com/pingcap/tidb/pull/13962 being merged and use the newer solution later.

Reply:
@breeswish Thanks for your advice! I'll wait for it.

Reply:
Feel free to pick up some other easy functions if you want :)